plugin/public/block.php

```diff
@@ -66,6 +67,25 @@ function add_block() {
 	);
 }
 
+/**
+ * Parses input string with dimension, e.g.:
+ *  - '100' -> '100px',
+ *  - '100px' -> '100px',
+ *  - '100%' -> '100%'.
+ * 
+ * @param $in Input string
+ * @return string If $in contains only digits append to it 'px' otherwise return unmodified $in
+ */
+function parse_dimension( $in ) {
+	$out = $in;
+
+	if ( ctype_digit( $in ) ) {
```

Comment:
Wouldn't `is_numeric` be stricter?

Reply:
Also, can't we do this before saving the setting to the DB as opposed to her on output? So that if you just put in `600` it would convert to `600px` and show that when you next open the settings?

Reply:
No - check PHP documentation & examples:
- ctype_digit() return true only when all characters are digits,
- is_numeric() return true when characters are numeric in general, i.e. also for: '1337e0', '1.9', etc.

I wrote this function for backward compatibility (for existing plugin users which may have defined some values in DB already). In general it may be good idea to convert this width value before saving it to DB but it is not resolving problem of backward compatibility.

Reply:
Ok, you're right about both. :)